internal/integration/integration_test.go

```diff
@@ -378,6 +378,41 @@ func (suite *SpokesReceivePackTestSuite) TestSpokesReceivePackWrongObjectSucceed
 	})
 }
 
+
+func (suite *SpokesReceivePackTestSuite) TestSpokesReceivePackIgnoreArgsSucceed() {
+	assert.NoError(suite.T(), chdir(suite.T(), suite.remoteRepo), "unable to chdir into our remote Git repo")
+	// Disable the `receive.fsckObjects option
+	require.NoError(suite.T(), exec.Command("git", "config", "receive.fsckObjects", "true").Run())
+	require.NoError(suite.T(), exec.Command("git", "config", "receive.fsck.missingEmail", "ignore").Run())
+	require.NoError(suite.T(), exec.Command("git", "config", "receive.fsck.badTagName", "ignore").Run())
+
+	assert.NoError(suite.T(), chdir(suite.T(), suite.localRepo), "unable to chdir into our local Git repo")
+
+	createBogusObjectAndPush(suite, func(suite *SpokesReceivePackTestSuite, err error, _ []byte) {
+		assert.NoError(
+			suite.T(),
+			err,
+			"unexpected error running the push with the custom spokes-receive-pack program; it should have succeeded since fsck args are ignored")
+	})
+}
+
+func (suite *SpokesReceivePackTestSuite) TestSpokesReceivePackMissingArgsFails() {
+	assert.NoError(suite.T(), chdir(suite.T(), suite.remoteRepo), "unable to chdir into our remote Git repo")
+	// Disable the `receive.fsckObjects option
+	require.NoError(suite.T(), exec.Command("git", "config", "receive.fsckObjects", "true").Run())
+	require.NoError(suite.T(), exec.Command("git", "config", "receive.fsck.missingEmail", "error").Run())
+	require.NoError(suite.T(), exec.Command("git", "config", "receive.fsck.badTagName", "error").Run())
+
+	assert.NoError(suite.T(), chdir(suite.T(), suite.localRepo), "unable to chdir into our local Git repo")
+
+	createBogusObjectAndPush(suite, func(suite *SpokesReceivePackTestSuite, err error, _ []byte) {
+		assert.Error(
+			suite.T(),
+			err,
+			"unexpected success running the push with the custom spokes-receive-pack program; it should have failed due to missing fsck args")
```

Comment:
I am not sure if you're testing the right thing here? The test is failing because you are pushing a bogus object not because missing fsck args?

Reply:
I am not sure what the implementation details of `createBogusObjectAndPush`are but from what I am observing setting fsck args to `ignore` makes the test succeed and setting them to `error` fails which is the expected behavior.

Reply:
Sorry, I misread the config setttings